### PR TITLE
Solution/grey noise1.1

### PR DESCRIPTION
**Pull request by punkrokk**
@v-prasadboke  Continued from https://github.com/Azure/Azure-Sentinel/pull/8931

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @punkrokk, I'll investigate the changes and update you about the same before 29 September, 2023.

**Comment by punkrokk** [comment]
@v-prasadboke Once you tell me everything is good, then I'll update the function zip file to master.

**Comment by v-prasadboke** [comment]
Ok Thanks.

File: Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseAPISentinelConnector/main.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                logging.error(e.response.text)
+            logging.error('Cannot upload indicators to Azure Sentinel, exiting.')
+            sys.exit(1)
+        
+        # Check for submission errors
+        if response.json().get('errors') != []:
+                logging.warning('Nonfatal error in submitting indicator. While a field failed, \n'  \
+                                'the rest of the indicator failed and we can continue.')
+                logging.warning('Error: ' + json.loads(response.json()).get('error'))
+        status_retry = 0
```
[the remark this answers is not in the document]
Comment (punkrokk) [reply Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseAPISentinelConnector/main.py]: fixed

**Comment by v-prasadboke** [comment]
Hello @punkrokk, manual instructions for deploying function app is missing from Data connector. Is it ok?
If not please follow this azure deploy on for manual deployment instructions of function app. [ **Line no 109** ]
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Bitglass/Data%20Connectors/Bitglass_API_FunctionApp.json

**Comment by punkrokk** [comment]
I think we can add manual deployment in v3.1, if that's ok 

I removed the python packages.

**Comment by punkrokk** [comment]
@v-prasadboke If we are good I'll update the function URL

File: Solutions/GreyNoiseThreatIntelligence/Data Connectors/azuredeploy_Connector_GreyNoiseAPISentinel_AzureFunction.json [hunk 2]
```diff
@@ -192,7 +193,8 @@
                           "CLIENT_ID": "[parameters('CLIENT_ID')]",
                           "CLIENT_SECRET": "[parameters('CLIENT_SECRET')]",
                           "GREYNOISE_CLASSIFICATIONS": "[parameters('GREYNOISE_CLASSIFICATIONS')]",
-                          "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/Azure/Azure-Sentinel/raw/db458a54839b084eac0e70bbe6e2a41f34f37e2b/Solutions/GreyNoiseThreatIntelligence/Data%20Connectors/GreyNoiseAPISentinelConn.zip"
+                          "GREYNOISE_LIMIT": "0",
+                          "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/Azure/Azure-Sentinel/raw/eda8de6774b9e82e9201babb5d1bba6c8cc80b04/Solutions/GreyNoiseThreatIntelligence/Data%20Connectors/GreyNoiseAPISentinelConn.zip"
```
Comment (v-prasadboke) [inline Solutions/GreyNoiseThreatIntelligence/Data Connectors/azuredeploy_Connector_GreyNoiseAPISentinel_AzureFunction.json:197]: Hello @punkrokk, Replace the link with "https://aka.ms/sentinel-GreyNoise-functionapp"

File: Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseConnector_UploadIndicatorsAPI.json [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        },{
+            "title": "5. Deploy the Threat Intellegence (Preview) Solution which includes the Threat Intelligence Upload Indicators API (Preview)",
+            "description": "See Microsoft Sentinel Content Hub for this Solution, and install it this Microsoft Sentinel instance."
+        },
+        {
+            "title": "6. Deploy the Azure Function",
+            "description": "Click the Deploy to Azure button.\n\n  [![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fa531176934d67267585713f73a777cedb3bbc4c1%2FSolutions%2FGreyNoise%2FData%2520Connectors%2Fazuredeploy_Connector_GreyNoiseAPISentinel_AzureFunction.json)\n\n Fill in the appropriate values for each parameter. **Be aware** that the only valid values for the **GREYNOISE_CLASSIFICATIONS** parameter are **malicious** and/or **unknown**, which must be comma separated. Do not bring in **<i>benign</i>**, as this will bring in millions of IPs which are known good and will likely cause many unwanted alerts."
```
Comment (v-prasadboke) [inline Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseConnector_UploadIndicatorsAPI.json:108]: Replace the azure deploy link with "https://aka.ms/sentinel-GreyNoise-azuredeploy"

**Comment by v-prasadboke** [comment]
Please repackage the solution after committing the changes.

File: Solutions/GreyNoiseThreatIntelligence/Data Connectors/azuredeploy_Connector_GreyNoiseAPISentinel_AzureFunction.json (hunk 2 above)
Comment (punkrokk) [reply Solutions/GreyNoiseThreatIntelligence/Data Connectors/azuredeploy_Connector_GreyNoiseAPISentinel_AzureFunction.json:197]: done

File: Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseConnector_UploadIndicatorsAPI.json (hunk 3 above)
Comment (punkrokk) [reply Solutions/GreyNoiseThreatIntelligence/Data Connectors/GreyNoiseConnector_UploadIndicatorsAPI.json]: done

**Comment by punkrokk** [comment]
@v-prasadboke all set and conflicts resolved.

**Comment by v-prasadboke** [comment]
Hello @punkrokk, Can you please provide write access to your branch/repo so that I can personally commit the required changes to this solution and get this PR closed as soon as possible.

File: Workbooks/WorkbooksMetadata.json [hunk 4]
```diff
@@ -6647,7 +6647,26 @@
       "domains": [
         "IT Operations"
       ]
-}
+},
```
Comment (v-prasadboke) [inline Workbooks/WorkbooksMetadata.json:6650]: A bracket "}" is missing above line 6650

Comment (punkrokk) [reply Workbooks/WorkbooksMetadata.json:6651]: I gave you write access and fixed the bracket.

**Comment by punkrokk** [comment]
> Hello @punkrokk, Can you please provide write access to your branch/repo so that I can personally commit the required changes to this solution and get this PR closed as soon as possible.

Gave you write access.

**Review by v-atulyadav — APPROVED** [review]

**Comment by manishkumar1991** [comment]
@punkrokk  We are seeing one vulnerability in your branch, Can you please delete the branch : BlueCycleOps:solution/GreyNoise1-1  
, As We can see that PR has been merged already . so if you no longer need this branch then please delete

**Pull request merged**